### ISSUE TITLE
yolov5 hard swish实现

### ISSUE THREAD

**Issue by facerless** [wang-xinyu/tensorrtx#183]
大佬，你好，目前hard swish是以插件形式实现的，感觉速度慢太多，同时插件只支持fp32模式。
我自己基于tensorrt api实现了hard swish，速度比插件要快不少。
                //hard swish
		Weights emptywts{ DataType::kFLOAT, nullptr, 0 };
		float *scval = reinterpret_cast<float*>(malloc(sizeof(float)));
		scval[0] = 1.0 / 6.0;
		Weights scale{ DataType::kFLOAT, scval, 1 };
		float *shval = reinterpret_cast<float*>(malloc(sizeof(float)));
		shval[0] = 0.5;
		Weights shift{ DataType::kFLOAT, shval, 1 };
		float *pval = reinterpret_cast<float*>(malloc(sizeof(float)));
		pval[0] = 1.0;
		Weights power{ DataType::kFLOAT, pval, 1 };
		auto clip = network->addActivation(*bn1->getOutput(0), ActivationType::kCLIP);
		clip->setAlpha(-3.0);
		clip->setBeta(3.0);
		auto sc = network->addScale(*clip->getOutput(0), ScaleMode::kUNIFORM, shift, scale, emptywts);
		auto hs = network->addElementWise(*sc->getOutput(0), *bn1->getOutput(0), ElementWiseOperation::kPROD);
		return hs;
除此之外，addBatchNorm2d中的eps应该设置为1e-5，而不是1e-3，pytorch里面默认使用的是1e-5。

**Comment by wang-xinyu** [comment]
@facerless 我这边model print出来看bn eps是0.001，你再确认下

这个hard swish你有对比过速度吗？有数据吗？或者你PR上来我测一下

**Comment by facerless** [comment]
我测过yolov5x 800，插件的耗时19ms，我这个17ms。

**Comment by facerless** [comment]
大佬，是我搞错了，SyncBatchNorm默认eps是1e-3，nn.BatchNorm才是1e-5

**Comment by wang-xinyu** [comment]
没用到SyncBatchNorm吧，就是nn.BatchNorm2d，我不知道model里面为什么是0.001

**Comment by facerless** [comment]
yolov5训练时启用了--sync-bn，就会把nn.BatchNorm2d替换成SyncBatchNorm，所以是0.001。我之前训练的分类网络，没用SyncBatchNorm，用的nn.BatchNorm2d，eps填入1e-5，结果才是完全一致的。

**Comment by wang-xinyu** [comment]
这样啊，明白了

**Comment by wang-xinyu** [comment]
@facerless 感谢你的hard swish idea， 我改成了 x * hard_sigmoid 
https://github.com/wang-xinyu/tensorrtx/commit/4980365b4dd15d301f591dd9e0b46b58fcfc1a12
我这边fp32会比之前plugin慢一点（yolov5s 608x608 7ms变为9ms），但是fp16应该会快，我测不了，没有fp16的显卡。等大家的反馈了。

**Comment by facerless** [comment]
我没测过fp32，我是在rtx2070上测的fp16

**Comment by OMG59E** [comment]
> 大佬，你好，目前hard swish是以插件形式实现的，感觉速度慢太多，同时插件只支持fp32模式。
> 我自己基于tensorrt api实现了hard swish，速度比插件要快不少。
> //hard swish
> Weights emptywts{ DataType::kFLOAT, nullptr, 0 };
> float _scval = reinterpret_cast<float_>(malloc(sizeof(float)));
> scval[0] = 1.0 / 6.0;
> Weights scale{ DataType::kFLOAT, scval, 1 };
> float _shval = reinterpret_cast<float_>(malloc(sizeof(float)));
> shval[0] = 0.5;
> Weights shift{ DataType::kFLOAT, shval, 1 };
> float _pval = reinterpret_cast<float_>(malloc(sizeof(float)));
> pval[0] = 1.0;
> Weights power{ DataType::kFLOAT, pval, 1 };
> auto clip = network->addActivation(*bn1->getOutput(0), ActivationType::kCLIP);
> clip->setAlpha(-3.0);
> clip->setBeta(3.0);
> auto sc = network->addScale(*clip->getOutput(0), ScaleMode::kUNIFORM, shift, scale, emptywts);
> auto hs = network->addElementWise(*sc->getOutput(0), *bn1->getOutput(0), ElementWiseOperation::kPROD);
> return hs;
> 除此之外，addBatchNorm2d中的eps应该设置为1e-5，而不是1e-3，pytorch里面默认使用的是1e-5。

这么搞会增大显存占用吧

**Comment by OMG59E** [comment]
> @facerless 感谢你的hard swish idea， 我改成了 x * hard_sigmoid
> [4980365](https://github.com/wang-xinyu/tensorrtx/commit/4980365b4dd15d301f591dd9e0b46b58fcfc1a12)
> 我这边fp32会比之前plugin慢一点（yolov5s 608x608 7ms变为9ms），但是fp16应该会快，我测不了，没有fp16的显卡。等大家的反馈了。

yolov5s，在2080Ti上fp16精度batch32前向1000次，平均每帧~1.63ms，同样参数，int8精度平均每帧~1.38ms，才快了0.3ms不到，正常么？

**Comment by pengfeidip** [comment]
> > @facerless 感谢你的hard swish idea， 我改成了 x * hard_sigmoid
> > [4980365](https://github.com/wang-xinyu/tensorrtx/commit/4980365b4dd15d301f591dd9e0b46b58fcfc1a12)
> > 我这边fp32会比之前plugin慢一点（yolov5s 608x608 7ms变为9ms），但是fp16应该会快，我测不了，没有fp16的显卡。等大家的反馈了。
> 
> yolov5s，在2080Ti上fp16精度batch32前向1000次，平均每帧~1.63ms，同样参数，int8精度平均每帧~1.38ms，才快了0.3ms不到，正常么？

@OMG59E  请问你在2080ti上对yolov5s进行fp16序列化速度如何？我这边测试非常慢
